core/domain/topic_services.py

```diff
@@ -294,11 +311,29 @@ def apply_change_list(topic_id, change_list):
                 elif (change.property_name ==
                       topic_domain.TOPIC_PROPERTY_LANGUAGE_CODE):
                     topic.update_language_code(change.new_value)
+                else:
+                    raise Exception('Invalid change dict.')
+            elif (change.cmd ==
+                  subtopic_page_domain.CMD_UPDATE_SUBTOPIC_PAGE_PROPERTY):
+                if (change.property_name ==
+                        subtopic_page_domain.SUBTOPIC_PAGE_PROPERTY_HTML_DATA):
+                    subtopic_pages = subtopic_page_services.update_html_data(
```

Comment:
Don't do this. Rather -- if the page is in newly_created_subtopics then edit it. Else get the subtopic page you need, update it, and add it to a modified_subtopic_pages list.

You could traverse the whole changelist once at the outset, and use that together with the existing next_subtopic_id, to figure out which existing subtopic pages are modified and fetch these all at once at the outset.

Reply:
Ok, yeah, will do that.

Reply:
Done, I am fetching the existing subtopic pages if a CMD_UPDATE_SUBTOPIC_PAGE_PROPERTY is encountered in apply_change_list.